### ISSUE TITLE
Sigentation fault after upgrade to github-pages-38

### ISSUE THREAD

**Issue by cathay4t** [github/pages-gem#156]
After upgrade from github-pages-35 to github-pages-38, `bundle exec jekyll serve` will fail
as `Segmentation fault`:

```
*** Error in `ruby': free(): invalid pointer: 0x0000000003d92520 ***
/home/fge/.gem/ruby/gems/celluloid-0.16.0/lib/celluloid.rb:478: [BUG] Segmentation fault
ruby 2.0.0p598 (2014-11-13) [x86_64-linux]

-- Control frame information -----------------------------------------------
c:0020 p:---- s:0081 e:000080 CFUNC  :require
c:0019 p:0242 s:0077 e:000076 TOP    /home/fge/.gem/ruby/gems/celluloid-0.16.0/lib/celluloid.rb:478 [FINISH]
c:0018 p:---- s:0075 e:000074 CFUNC  :require
c:0017 p:0007 s:0071 e:000070 TOP    /home/fge/.gem/ruby/gems/listen-2.10.1/lib/listen.rb:1 [FINISH]
c:0016 p:---- s:0069 e:000068 CFUNC  :require
c:0015 p:0009 s:0065 e:000064 METHOD /home/fge/.gem/ruby/gems/jekyll-watch-1.2.1/lib/jekyll/watcher.rb:26
c:0014 p:0028 s:0060 e:000059 METHOD /home/fge/.gem/ruby/gems/jekyll-watch-1.2.1/lib/jekyll/watcher.rb:7
c:0013 p:0031 s:0053 e:000052 METHOD /home/fge/.gem/ruby/gems/jekyll-2.4.0/lib/jekyll/commands/build.rb:67
c:0012 p:0115 s:0048 e:000047 METHOD /home/fge/.gem/ruby/gems/jekyll-2.4.0/lib/jekyll/commands/build.rb:37
c:0011 p:0049 s:0043 e:000042 BLOCK  /home/fge/.gem/ruby/gems/jekyll-2.4.0/lib/jekyll/commands/serve.rb:25 [FINISH]
c:0010 p:---- s:0039 e:000038 CFUNC  :call
c:0009 p:0012 s:0034 e:000033 BLOCK  /home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/command.rb:220 [FINISH]
c:0008 p:---- s:0031 e:000030 CFUNC  :each
c:0007 p:0044 s:0028 e:000027 METHOD /home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/command.rb:220
c:0006 p:0082 s:0023 e:000022 METHOD /home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/program.rb:42
c:0005 p:0037 s:0017 e:000016 METHOD /home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary.rb:19
c:0004 p:0104 s:0012 E:000fe8 TOP    /home/fge/.gem/ruby/gems/jekyll-2.4.0/bin/jekyll:18 [FINISH]
c:0003 p:---- s:0010 e:000009 CFUNC  :load
c:0002 p:0118 s:0006 E:0017b8 EVAL   /home/fge/bin/jekyll:23 [FINISH]
c:0001 p:0000 s:0002 E:000308 TOP    [FINISH]

-- Ruby level backtrace information ----------------------------------------
/home/fge/bin/jekyll:23:in `<main>'
/home/fge/bin/jekyll:23:in `load'
/home/fge/.gem/ruby/gems/jekyll-2.4.0/bin/jekyll:18:in `<top (required)>'
/home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary.rb:19:in `program'
/home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/program.rb:42:in `go'
/home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `execute'
/home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `each'
/home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `block in execute'
/home/fge/.gem/ruby/gems/mercenary-0.3.5/lib/mercenary/command.rb:220:in `call'
/home/fge/.gem/ruby/gems/jekyll-2.4.0/lib/jekyll/commands/serve.rb:25:in `block (2 levels) in init_with_program'
/home/fge/.gem/ruby/gems/jekyll-2.4.0/lib/jekyll/commands/build.rb:37:in `process'
/home/fge/.gem/ruby/gems/jekyll-2.4.0/lib/jekyll/commands/build.rb:67:in `watch'
/home/fge/.gem/ruby/gems/jekyll-watch-1.2.1/lib/jekyll/watcher.rb:7:in `watch'
/home/fge/.gem/ruby/gems/jekyll-watch-1.2.1/lib/jekyll/watcher.rb:26:in `build_listener'
/home/fge/.gem/ruby/gems/jekyll-watch-1.2.1/lib/jekyll/watcher.rb:26:in `require'
/home/fge/.gem/ruby/gems/listen-2.10.1/lib/listen.rb:1:in `<top (required)>'
/home/fge/.gem/ruby/gems/listen-2.10.1/lib/listen.rb:1:in `require'
/home/fge/.gem/ruby/gems/celluloid-0.16.0/lib/celluloid.rb:478:in `<top (required)>'
/home/fge/.gem/ruby/gems/celluloid-0.16.0/lib/celluloid.rb:478:in `require'

-- C level backtrace information -------------------------------------------
<command hang and require kill -9>
````
OS is RHEL 7.1 with `ruby-2.0.0.598-25.el7_1.x86_64` and `rubygem-bundler-1.3.1-3.el7.noarch`.

Tried new install on debian 8.1(ruby 2.1.5+deb8u1), got the same error.

**Comment by parkr** [comment]
Did you try re-installing celluloid and listen?

**Comment by cathay4t** [comment]
> Tried new install on debian 8.1(ruby 2.1.5+deb8u1), got the same error.

I tried to remove ~/.gem folder completely and reinstall. Still the same.

**Comment by spraints** [comment]
Can you try using github-pages-37?

**Comment by cathay4t** [comment]
@spraints github-pages-37 works well.

**Comment by robin850** [comment]
Well, looking at the [diff](https://github.com/github/pages-gem/compare/v37...v38) between these two versions, it may be related to vmg/redcarpet#494. The problem was that I wrote a function which wrongly managed memory but Vicent fixed it in version 3.3.2. Could you try with this version @cathay4t please ?

**Comment by cathay4t** [comment]
@robin850 It fixed this problem:
```
gem install redcarpet --version 3.3.2
gem cleanup redcarpet
gem uninstall redcarpet --version 3.3.1
jekyll serve
```

Thanks for the help.

**Comment by robin850** [comment]
Nice! Thanks for reporting ; this can be closed once #149 is merged then!

**Comment by parkr** [comment]
Please close this issue.
